### DISCUSSION TITLE
Low memory usage help for thin client

### DISCUSSION THREAD

**Discussion by jv-oliveira** [FreeRDP/FreeRDP#7518] [Q&A]
Hi,
I'd be grateful if you guys could help me with some tips for setting up the library to minimize memory usage.

I'm POC developing a RDP thin client to be run on an armv7 custom board with limited RAM and embedded GNU/Linux. I'm using Qt5 and FreeRDP 2.4.1. The code can be run on my x86 desktop too.

The memory usage seems to be ok in simple use cases, but it drastically increases if too many events happen in the server and don't free it afterward. I'd like to tune it to the minimum memory usage possible and make it have a limit for resources without often crashing.

I am using the freerdp_client_* functions and use the settings macros from settings.h to set up the library.

**Comment by akallabeth** [comment]
No easy answer here.
This depends a lot on how you connect to the server (e.g. do you use `gfx`, windows 7 style `rfx` or old style `bitmap/gdi` mode) and what you require (e.g. play video/animations, have some small nearly static form to display, ...)

Usually the best option is to use the switches set by `/network:auto` and let the server choose the best supported codec.
As for memory consumption, you don´t have many options with `gfx` (if your server supports h264 it will not use buffers anyway) and for legacy you just don´t enable bitmap/offscreen/... chaches (but that might break some connections)

Anything more will require you to test and analyze where you need to `trim` memory.
As you also mentioned you´re using FreeRDP as a library check against `xfreerdp` behavior, so you don´t have a memory leak in your client by accident.
